### PR TITLE
feat: Adds e2e tests for chain/schedule policy and bump ScheduledAutoscaler to Alpha

### PR DESCRIPTION
**Pull request by indexjoseph**
**What type of PR is this?**

 /kind feature

**What this PR does / Why we need it**:

This PR is a follow-up of #3929 

- Adds e2e test for schedule policy to verify active/inactive schedules
- Adds e2e test for chain policy to verify fall through and back-to-back scheduling
- Changes Scheduled Autoscaler feature gate from Dev to Alpha

**Which issue(s) this PR fixes**:

Towards #3008, #3686

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by zmerlynn** [comment]
Please move the feature to Alpha so it's tested in CI! :)

https://github.com/googleforgames/agones/blob/main/pkg/util/runtime/features.go#L91-L95

**Review by zmerlynn — COMMENTED** [review]
The test LGTM. I'd like to see it pass in CI before a maintainer approves it - see my note above about graduating it to Alpha, which will get it tested here.

**Comment by indexjoseph** [comment]
Hmm, it looks like `submit-e2e-test-cloud-build` may be failing due to a flake, this is the same error that I was receiving yesterday but triggering CI again it passed.

File: pkg/util/runtime/features.go
Comment (igooch) [inline pkg/util/runtime/features.go:139]: I realize this isn't part of your PR, but could you please move `FeatureRollingUpdateFix` so that it's in alphabetical order?

**Review by igooch — CHANGES_REQUESTED** [review]
Tests overall look good, just a few small changes needed.

File: build/Makefile
Comment (indexjoseph) [reply build/Makefile:732]: I don't quite understand, should I restore the `install.yaml` file to the original version?

File: pkg/util/runtime/features.go
Comment (indexjoseph) [reply pkg/util/runtime/features.go:139]: Sounds good!

File: build/Makefile
Comment (igooch) [reply build/Makefile:732]: Remove the added 
```suggestion
```

File: build/Makefile
Comment (igooch) [inline build/Makefile:740]:
```suggestion
```

File: test/e2e/fleetautoscaler_test.go
Comment (igooch) [inline test/e2e/fleetautoscaler_test.go:1692]:
```suggestion
	assert.Nil(t, err)
```

**Comment by igooch** [comment]
If you re-run `~/agones/build$ make gen-install` and push the changes the tests should pass.

**Comment by indexjoseph** [comment]
> If you re-run `~/agones/build$ make gen-install` and push the changes the tests should pass.

I'm trying to configure my dev environment, I've recently upgraded to a Mac w/ an Apple silicon chip and I've been having trouble trying to get Docker working. I'll rerun `make gen-install` on a my other machine here soon.

**Comment by zmerlynn** [comment]
> > If you re-run `~/agones/build$ make gen-install` and push the changes the tests should pass.
> 
> I'm trying to configure my dev environment, I've recently upgraded to a Mac w/ an Apple silicon chip and I've been having trouble trying to get Docker working. I'll rerun `make gen-install` on my other machine here soon.

You can also just use the `helm` command that's in the Makefile target and skip Docker here, if you can install helm (which may be easier than Docker).

File: site/content/en/docs/Guides/feature-stages.md
Comment (igooch) [inline site/content/en/docs/Guides/feature-stages.md:43]: Apologies for not being clearer on my previous comment, this should go down on line 57. The code block that will be published with version 1.43 starts with `{{% feature publishVersion="1.43.0" %}}` on line 44 and is closed by `{{% /feature %}}` on line 59.

Comment (indexjoseph) [reply site/content/en/docs/Guides/feature-stages.md:43]: Oh I see, np.

**Review by igooch — APPROVED** [review]
LGTM, good work!

**Pull request merged**
